### ISSUE TITLE
About the batch size in training detection network

### ISSUE THREAD

**Issue by yw155** [weiliu89/caffe#78]
The batch size is set to 32 in the paper. Is this size indicates the number of input images?

In the SSD method, the network can generate a lot of detection windows and using these windows to compute the training loss. Since the number of windows is large enough, is it can further reduce the batch size in the training, even to 1? What is the effect of batch size to the training?

Thank you!

**Comment by weiliu89** [comment]
32 is the number of images in a batch. Although there is a lot of sampled patches from an image, we only select one from each image.

**Comment by yw155** [comment]
I got that. Thank you!

**Comment by conner99** [comment]
I set a batch_size of 4

and add this to ssd_pascal.py, because the post in **https://github.com/BVLC/caffe/issues/430#issuecomment-43921545**
```
#if batch_size is divided by X then max_iter and stepsize should multiply by X
max_iter = 60000 * (32 / batch_size)
stepsize = 40000 * (32 / batch_size)
```
